### DISCUSSION TITLE
Extend DefineStoreOptionsBase typescript no unused vars

### DISCUSSION THREAD

**Discussion by DaceExMachina** [vuejs/pinia#2176] [Help and Questions]
Hi, I'm trying to implement an optional plugin for a pinia store similar to [this suggestion](https://github.com/vuejs/pinia/issues/1128).

I'm unfortunately running into a couple type errors that I'm not sure how to resolve.  It seems that when I override `DefineStoreOptonsBase` as per [this doc](https://pinia.vuejs.org/core-concepts/plugins.html#typing-new-creation-options), and then add the 'plugins' property to the definsStore options, I get a "ESLint: 'Store' is defined but never used.(@typescript-eslint/no-unused-vars)" warning because I do not need them. But when I remove `<S,Store>` from DefineStoreOptionsBase, I get a different warning stating: "TS2428: All declarations of 'DefineStoreOptionsBase' must have identical type parameters."

It states in the documentation that everything should able to be typed safely without disabling any type warnings but I cannot seem to figure out how to deal with this case and I couldn't find promising results while searching for answers. Any suggestions would be appreciated.

```
import 'pinia'

export enum PluginsEnum {
	MockData = 'mockData',
}

type PluginId = typeof PluginsEnum[keyof typeof PluginsEnum]

declare module 'pinia' {
	export interface DefineStoreOptionsBase<S, Store> {
		plugins?: PluginId[]
	}
}
```

**Comment by posva** [comment]
Unfortunately, in this case, you need to tell eslint to ignore the error
